### PR TITLE
refactor run_in_windows_bash

### LINKED ISSUE #4680: [Conan 2.0] Study removal of `--login` by default at `run_in_windows_bash`
Valuable feedback here:
https://github.com/conan-io/conan/issues/4304

Study if remove by default the ``--login`` in Conan 2.0

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

To prepare better for https://github.com/conan-io/conan/pull/11066

Close https://github.com/conan-io/conan/issues/4680

### CODE REVIEW

File: conan/tools/cmake/cmake.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def _build(self, build_type=None, target=None, cli_args=None, build_tool_args=No
         if cmd_line_args:
             args += ['--'] + cmd_line_args
 
-        arg_list = [args_to_string([bf]), build_config, args_to_string(args)]
```
Comment (lasote) [inline conan/tools/cmake/cmake.py:125]: Risk of 🍔 ?

File: conan/tools/env/environment.py [hunk 2]
```diff
@@ -13,34 +13,35 @@ class _EnvVarPlaceHolder:
     pass
 
 
-def environment_wrap_command(env_filenames, cmd, subsystem=None, cwd=None):
-    assert env_filenames
+def environment_wrap_command(env_filenames, env_folder, cmd, subsystem=None, accept=None):
```
Comment (lasote) [inline conan/tools/env/environment.py]:
```suggestion
def environment_wrap_command(env_filenames, env_folder, cmd, subsystem=None, accepted_extensions=None):
```

File: conans/client/subsystems.py [hunk 3]
```diff
@@ -89,6 +90,14 @@ def escape_windows_cmd(command):
 
 
 def deduce_subsystem(conanfile, scope):
+    """ used by:
+    - EnvVars: to decide is using :  ; as path separator, translate paths to subsystem
```
Comment (lasote) [inline conans/client/subsystems.py]:
```suggestion
    - EnvVars: to decide if using :  ; as path separator, translate paths to subsystem
```

**Review by lasote — APPROVED** [review]

File: conan/tools/cmake/cmake.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/cmake/cmake.py:125]: Actually, that ``args_to_string`` calls ``subprocess.list2cmdline`` which is an undocumented, apparently internal ``subprocess`` function. I don't think we should be calling that at all. I prefer to risk the 🍔 than depending on that.

Anyone can provide some evidence about ``subprocess.list2cmdline`` being ok to be used?

Comment (memsharded) [reply conan/tools/cmake/cmake.py:125]: https://github.com/conan-io/conan/issues/11358

**Pull request merged**